src/perception/tag_detector/tag_detector.processing.cpp

```diff
@@ -89,7 +92,45 @@ namespace mrover {
           tag.hitCount = std::clamp(tag.hitCount + 1, 0, mMaxHitCount);
           tag.id = id;
           tag.imageCenter = std::accumulate(mCorners[i].begin(), mCorners[i].end(), cv::Point2f{}) / static_cast<float>(mCorners[i].size());
-          tag.tagInCam = getFidInCamFromPixel(msg, std::lround(tag.imageCenter.x), std::lround(tag.imageCenter.y));
+
+          // find extremes of detected tag for checking later
+          float leftmost = std::numeric_limits<float>::max();
+          float rightmost = 0.0;
+          float uppermost = std::numeric_limits<float>::max();
+          float lowermost = 0.0;
+
+          for (size_t j = 0; j < mCorners[i].size(); ++j) {
+            if (mCorners[i][j].x < leftmost) leftmost = mCorners[i][j].x;
+            if (mCorners[i][j].x > rightmost) rightmost = mCorners[i][j].x;
+            if (mCorners[i][j].y > lowermost) lowermost = mCorners[i][j].y;
+            if (mCorners[i][j].y < uppermost) uppermost = mCorners[i][j].y;
+          }
+
+          R3 acc;
+          acc.setZero();
+
+          int numValid = 0;
+          
+          for (int x = lround(tag.imageCenter.x) - PADDING; x < lround(tag.imageCenter.x) + PADDING; ++x) {
```

Comment:
I'm wondering if a better strategy is to start at the tag center and work our way outward radially along each axis to find a valid depth. I don't think we really want to average all the points on the tag because it will needlessly perturb the results when the center px does exist but there is other sparsity

Reply:
assuming the probability that any given px depth is accurate in the depth image is iid (its not), averaging all the px increases the probability that your results are consistently not so good.